src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -147,6 +149,36 @@ private void processRequests(HttpServerRequest httpServerRequest) {
 
                 break;
 
+            //create a sink endpoint and initialize consumer
+            case CREATE:
+                SinkBridgeEndpoint<?,?> sink = new HttpSinkBridgeEndpoint<>(this.vertx, this.bridgeConfigProperties);
+                sink.closeHandler(s -> {
+                    this.endpoints.get(httpServerRequest.connection()).getSinks().remove(s);
+                });
+
+                sink.open();
+
+                //add sink to list
+                this.endpoints.get(httpServerRequest.connection()).getSinks().add(sink);
+
+                sink.createConsumer(new HttpEndpoint(httpServerRequest));
+                break;
+
+            case SUBSCRIBE:
+
+                String instanceId = PathParamsExtractor.getConsumerSubscriptionParams(httpServerRequest).get("instance-id");
+
+                List<SinkBridgeEndpoint> sinkList = this.endpoints.get(httpServerRequest.connection()).getSinks();
+
+                //consumer lookup
+                final SinkBridgeEndpoint subscribeSink = sinkList.stream()
+                        .filter(sink1 -> sink1.consumerInstanceId.equals(instanceId))
+                        .findFirst()
+                        .orElseThrow(() -> new RuntimeException("No consumer found with given instance-id"));
+
+                subscribeSink.subscribeToTopic(new HttpEndpoint(httpServerRequest));
```

Comment:
Same as above, calling the `handle` method on the `subscribeSink` (call it just `sink`) and leaving it to do the work.